backend/src/main/java/cz/muni/fi/pa165/pneuservis/backend/dao/CustomerDaoImpl.java

```diff
@@ -0,0 +1,108 @@
+package cz.muni.fi.pa165.pneuservis.backend.dao;
+
+import cz.muni.fi.pa165.pneuservis.backend.entity.Customer;
+
+import javax.persistence.EntityManager;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * @author Zuzana Žufanová, zufinka@mail.muni.cz
+ */
+public class CustomerDaoImpl implements CustomerDao{
+    private EntityManager em;
+
+    public CustomerDaoImpl (EntityManager em){
+        this.em = em;
+    }
+
+    @Override
+    public Customer findById(Long id) {
+        if (id == null){
+            throw new IllegalArgumentException("ID is null.");
+        }
+        return em.find(Customer.class, id);
+    }
+
+    @Override
+    public List<Customer> findByName(String name, String surname) {
+        if (name == null || surname == null){
+            throw new IllegalArgumentException("Fist name or surname is null.");
+        }
+        return em.createQuery("SELECT c FROM CustomerTable c WHERE c.name = :name AND c.surname = :surname",
```

Comment:
This should be the entity class name, not the table name ("Customer" instead of "CustomerTable").